contracts/pcv/morpho/MorphoCompoundPCVDeposit.sol

```diff
@@ -0,0 +1,384 @@
+/// then withdrawing at least 1 block later. That way, interest accrues.
+/// This is not a Morpho specific issue. Compound rounds in the protocol's favor.
+/// The issue is caused by constraints inherent to solidity and the EVM.
+/// There are no floating point numbers, this means there is precision loss,
+/// and protocol engineers are forced to choose who to round in favor of.
+/// Engineers must round in favor of the protocol to avoid deposits of 0 giving
+/// the user a balance.
+contract MorphoCompoundPCVDeposit is PCVDeposit, ReentrancyGuard {
+    using SafeERC20 for IERC20;
+    using SafeCast for *;
+
+    /// ------------------------------------------
+    /// ----------------- Event ------------------
+    /// ------------------------------------------
+
+    /// @notice emitted when the PCV Oracle address is updated
+    event PCVOracleUpdated(address oldOracle, address newOracle);
+
+    /// ------------------------------------------
+    /// ---------- Immutables/Constant -----------
+    /// ------------------------------------------
+
+    /// @notice reference to the COMP governance token
+    /// used for recording COMP rewards type in Harvest event
+    address public constant COMP = 0xc00e94Cb662C3520282E6f5717214004A7f26888;
+
+    /// @notice reference to the lens contract for morpho-compound v2
+    address public immutable lens;
+
+    /// @notice reference to the morpho-compound v2 market
+    address public immutable morpho;
+
+    /// @notice reference to underlying token
+    address public immutable token;
+
+    /// @notice cToken in compound this deposit tracks
+    /// used to inform morpho about the desired market to supply liquidity
+    address public immutable cToken;
+
+    /// ------------------------------------------
+    /// ------------- State Variables -------------
+    /// ------------------------------------------
+
+    /// @notice track the last amount of PCV recorded in the contract
+    /// this is always out of date, except when accrue() is called
+    /// in the same block or transaction. This means the value is stale
+    /// most of the time.
+    uint256 public lastRecordedBalance;
+
+    /// @notice reference to the PCV Oracle. Settable by governance
+    /// if set, anytime PCV is updated, delta is sent in to update liquid
+    /// amount of PCV held
+    /// not set in the constructor
+    address public pcvOracle;
+
+    /// @param _core reference to the core contract
+    /// @param _cToken cToken this deposit references
+    /// @param _underlying Token denomination of this deposit
+    /// @param _morpho reference to the morpho-compound v2 market
+    /// @param _lens reference to the morpho-compound v2 lens
+    constructor(
+        address _core,
+        address _cToken,
+        address _underlying,
+        address _morpho,
+        address _lens
+    ) CoreRef(_core) ReentrancyGuard() {
+        if (_underlying != address(Constants.WETH)) {
```

Comment:
Why's `(_underlying != address(Constants.WETH))` there? I thought the underlying was USDC always?

Reply:
if we ever use this deposit for eth, the ceth contract has no underlying function and will revert.

Reply:
USDC and DAI, and in the future, could be other tokens.